pkg/crypto/crypto_test.go

```diff
@@ -108,3 +109,69 @@ func decryptConfigMap(configMap string, keyPath string, noncePath string) string
 
 	return string(plaintextBytes)
 }
+
+func TestDecryptConfigMapWrongkey(t *testing.T)  {
+        
+	testString := "Sample String to Test Symmetric Encryption"
+
+        dir1, err := ioutil.TempDir("", "TestEncryption-2")
+        if err != nil {
+                t.Error(err)
+        }
+        defer os.RemoveAll(dir1)
+
+        keyPath := dir1 + "symmKeyFile"
+        wrongkeyPath := dir1 + "wrongsymmKeyFile"
+
+        key := make([]byte, 32)
+        if _, err := io.ReadFull(rand.Reader, key); err != nil {
+                t.Error(err)
+        }
+        wrongkey := make([]byte, 32)
+        if _, err := io.ReadFull(rand.Reader, wrongkey); err != nil {
+                t.Error(err)
+        }
+
+        //Encode the Key
+        keyEnc := b64.StdEncoding.EncodeToString(key)
+        //encode wrong key
+        wrongkeyEnc := b64.StdEncoding.EncodeToString(wrongkey)
+
+        err = ioutil.WriteFile(keyPath, []byte(keyEnc), 0644)
+        if err != nil {
+                t.Error(err)
+        }
+        err=ioutil.WriteFile(wrongkeyPath, []byte(wrongkeyEnc), 0644)
+	if err != nil {
+                t.Error(err)
+        }
+
+        noncePath := dir1 + "nonceFile"
+        nonce := make([]byte, 12)
+        if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+                t.Error(err)
+        }
+
+        //Encode the nonce
+        nonceEnc := b64.StdEncoding.EncodeToString(nonce)
+
+        err = ioutil.WriteFile(noncePath, []byte(nonceEnc), 0644)
+        if err != nil {
+                t.Error(err)
+        }
+
+        encConfigMap, err := EncryptConfigMap([]byte(testString), keyPath, noncePath)
+
+        if err != nil {
+                t.Errorf("Failed to encrypt configMap %v", err)
+        }
+
+        wrongdecryptedConfigMap := decryptConfigMap(encConfigMap, wrongkeyPath, noncePath)
```

Comment:
I'm just wondering if this code is about testing the utility function decryptConfigMap which is not really needed I feel.

Still, if we are writing then - the better approach would be to return the error message from the function instead of panicking and compare the error with the expected error.

Reply:
Thanks for the suggestions. decryptConfigMap enters into panic for each error case in existing design. Please suggest if it would be feasible to replace the panics to t.Error; so that error cases are returned rather than being exited.

Reply:
Please squash the two commits into a single one and re-push

Reply:
> 
> 
> Please squash the two commits into a single one and re-push

Thanks for the suggestions. done now.